apps/frontend/core-dapp/src/features/trade/close-trade/CloseTradeStore.ts

```diff
@@ -0,0 +1,56 @@
+import { BigNumber } from 'ethers'
+import { makeAutoObservable } from 'mobx'
+import { validateStringToBN } from 'prepo-utils'
+import { RootStore } from '../../../stores/RootStore'
+import { MarketEntity } from '../../../stores/entities/MarketEntity'
+import { PositionEntity } from '../../../stores/entities/Position.entity'
+
+export class CloseTradeStore {
+  input: { type: 'max' } | { type: 'input'; value: string } = {
+    type: 'input',
+    value: '',
+  }
+  constructor(private root: RootStore) {
+    makeAutoObservable(this, {}, { autoBind: true })
+  }
+
+  // selected market and positions, for ease of accessing
+  // so we don't have to keep doing this.root.tradeStore...
+  // and can just use this.selectedMarket
+  get selectedMarket(): MarketEntity | undefined {
+    return this.root.tradeStore.selectedMarket
+  }
```

Comment:
```suggestion
  private get selectedMarket(): MarketEntity | undefined {
    return this.root.tradeStore.selectedMarket
  }
```

I would remove the comment, I think it's self-explanatory. And maybe private?

Reply:
addressed this here https://github.com/prepo-io/prepo-monorepo/pull/1224/files#diff-3b910fb66dd8dc572d9efb9c7fb45569eb22c304ccbbe919076a13b4600e8b72L66-L68